#164859756 users can follow each other

jmusila opened the pull request:
### What does this PR do?
- Adds user following and unfollowing functionality

#### Description of Task to be completed?

- [x] write unit tests
- [x] add user follow each other functionality
- [x] add user unfollow another user functionality
- [x] add user view all followers functionality
- [x] add user view all followees functionality

#### How should this be manually tested?

- Clone the repo
  `$ git clone https://github.com/deferral/ah-django`

- Create a virtual environment with virtualenv
  `$ virtualenv venv`

- Configure the database to work with the application
  `$ python manage.py makemigrations`
   `$ python manage.py migrate`

- Run the application
  `$ python manage.py runserver`
 - Testing Routes on Postman
```
Follow a user
- /api/profiles/<username>/follow

Unfollow a user
- /api/profiles/<username>/unfollow

List following
- /api/profiles/<username>/following

List Followers
- /api/profiles/<username>/followers
```

#### Any background context you want to provide?

#### What are the relevant pivotal tracker stories?
#164859756 

Screenshots (if appropriate)
![Insomnia – http:--127 0 0 1:8000-api-profiles-cowmammal-follow_168](https://user-images.githubusercontent.com/42172425/55645135-69c01e80-57e0-11e9-82d6-648120a9fe99.png)
![Insomnia – http:--127 0 0 1:8000-api-profiles-cowmammal-followers_169](https://user-images.githubusercontent.com/42172425/55645149-73498680-57e0-11e9-824d-d3af1bd79653.png)
![Insomnia – http:--127 0 0 1:8000-api-profiles-cowmammal-unfollow_170](https://user-images.githubusercontent.com/42172425/55645161-7c3a5800-57e0-11e9-95c6-d2f3335bc745.png)
![Insomnia – http:--127 0 0 1:8000-api-profiles-cowmammal-following_171](https://user-images.githubusercontent.com/42172425/55645173-852b2980-57e0-11e9-8277-698a5b0a040d.png)

mugoh commented on authors/apps/followers/views.py:
```diff
@@ -0,0 +1,136 @@
+    #     return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+class FollowersView(ListCreateAPIView):
```
Since we are not implementing the `post` method here,and having a serializer class that inherits `rest_framework.serializers.Baseserializer`
 `class FollowerSerializer(serializers.BaseSerializer): ` 

```suggestion
class FollowersView(ListApiView):
```
This will, on attempt to send a post request to `../followers/`, help us avoid
![ser_err](https://user-images.githubusercontent.com/44059971/55674853-b11edb80-58c2-11e9-97a2-78009703e982.jpg)

mugoh commented on authors/apps/followers/views.py:
```diff
@@ -0,0 +1,136 @@
+                        status.HTTP_404_NOT_FOUND)
+
+
+class RetrieveFollowing(ListCreateAPIView):
```
Look into this line too

mugoh reviewed (commented):
```NotImplementedError: `to_internal_value()` must be implemented```
This is raised when testing some of the edge cases. More on this in the inline comments.
But it's good that the the routes work as described.

jmusila committed:
164859756 feat(follow-each-other): users should be able to follow each other

- a user should be able to follow another user
- a user should be able to unfollow followed user
- a user should be able to see list of followers
- a user should be able to see list of followees

[Delivers #164859756]: ft-followers-164859756
